src/utils/getDisplayName.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import { t } from '@nextcloud/l10n'
+
+import { ATTENDEE } from '../constants.js'
+
+export const getDisplayNameWithFallback = function(displayName: string, source: string): string {
+	if (displayName?.trim()) {
+		return displayName.trim()
+	}
+
+	if ([ATTENDEE.ACTOR_TYPE.GUESTS, ATTENDEE.ACTOR_TYPE.EMAILS].includes(source)) {
+		return t('spreed', 'Guest')
+	}
+
+	if (source === ATTENDEE.ACTOR_TYPE.DELETED_USERS) {
+		return t('spreed', 'Deleted user')
+	}
+
+	// Fallback to 'Deleted user': should not happen, but can not be empty either
+	return t('spreed', 'Deleted user')
+}
```

Comment:
can be merged into one return

Reply:
Logic is different in two cases, but for the sake of reducing code keeping only fallback return with comment in code source